src/lib/client/index.ts

```diff
@@ -1,114 +1,78 @@
-import { generateQuery } from '../../integrations/cdc-ai-agent.api.js';
-import { CdcAiAgentResponse } from '../../integrations/cdc-ai-agent.interfaces.js';
+import { generateQuery, Query } from '../../integrations/cdc-ai-agent.api';
+import { CdcAiAgentResponse } from '../../integrations/cdc-ai-agent.interfaces';
 
 export interface CdcAiAgentClient {
   agent: Query;
 }
-
-export type Query = {
-  /**
-   * Sends a query to the CDC Agent AI Service and fetches the AI-generated response.
-   *
-   * @param {string} query - The query string to be sent to the CDC Agent AI Service (e.g., questions or commands related to blockchain).
-   * @returns {Promise<{ message: string; object: any }>} A promise that resolves to an object containing the AI-generated message and related data.
-   *
-   * @example
-   * const client = createClient({ openAI: { apiKey: 'YOUR_OPEN_AI_API_KEY' }, chain: { rpc: 'CHAIN_RPC' }, explorer: { apiKey: 'EXPLORER_API_KEY' }, wallet: { mnemonic: 'YOUR_WALLET_MNEMONIC' } });
-   * async function sendQuery() {
-   *   try {
-   *     const response = await client.agent.generateQuery('What is the latest block?');
-   *     console.log('CDC Agent AI Response:', response.message);
-   *   } catch (e) {
-   *     console.error('Error fetching CDC Agent AI response:', e);
-   *   }
-   * }
-   * sendQuery();
-   */
-  generateQuery: (query: string) => Promise<CdcAiAgentResponse>;
-};
-
 /**
- * Configuration parameters for creating a CDC Agent AI client instance.
+ * Client params for creating a CDC AI Agent client instance.
  *
  * @interface
- * @property {object} openAI - Configuration for OpenAI API access, including the API key.
- * @property {object} chain - Configuration for blockchain specifics such as chain ID, name, and RPC URL.
- * @property {object} explorer - Configuration for blockchain explorer API access, including the base URL and API key.
+ * @property {object} openAI - The API key for OpenAI and the model to use. Defaults to gpt-4-turbo if not provided.
+ * @property {number} chainId - The chain number:
+ * - 25: cronos
+ * - 338: cronos-testnet
+ * - 388: cronos-zkevm
+ * - 282: cronos-zkevm-testnet
+ * @property {object} explorerKeys - The API keys for the respective explorers.
+ * @property {string} customRPC - Optional custom RPC URL. If not provided, the default RPC for the chain will be used.
  */
-export interface ClientConfig {
-  /** Configuration for OpenAI API access */
+export interface QueryOptions {
   openAI: {
     apiKey: string;
+    model?: string;
   };
-
-  /** Configuration for blockchain specifics */
-  chain: {
-    id: number;
-    name: string;
-    rpc: string;
+  chainId: number;
+  explorerKeys: {
+    cronosMainnetKey?: string;
+    cronosTestnetKey?: string;
+    cronosZkEvmKey?: string;
+    cronosZkEvmTestnetKey?: string;
   };
+  signerAppUrl?: string;
+  customRPC?: string;
+  context?: QueryContext[];
```

Comment:
@leejw51crypto Here is the context changes

Reply:
response also return update context:[]
```
Query Result:
{
  "status": "success",
  "hasErrors": false,
  "results": [
    {
      "status": "Success",
      "function": "getLatestBlock",
      "message": "Latest block height: 289874",
      "data": {
        "blockHeight": 289874,
        "timestamp": "2024-10-07T05:37:21.000Z"
      }
    }
  ],
  "context": [
    {
      "role": "user",
      "content": "get latest block"
    },
    {
      "role": "assistant",
      "content": "[{\"status\":\"Success\",\"function\":\"getLatestBlock\",\"message\":\"Latest block height: 289874\",\"data\":{\"blockHeight\":289874,\"timestamp\":\"2024-10-07T05:37:21.000Z\"}}]"
    }
  ]
}
```

Reply:
return context in response, so that user use it in next query